geopandas/io/file.py

```diff
@@ -384,7 +419,7 @@ def _read_file_fiona(
             else:
                 f_filt = features
             # get list of columns
-            columns = list(features.schema["properties"])
+            columns = columns or list(features.schema["properties"])
```

Comment:
Using the user specified list instead of the one of the schema ensures we follow the column order as specified by the user (this seems to be what pyogrio does, although its docs say otherwise)

Reply:
```suggestion
            columns = columns if columns is not None else list(features.schema["properties"])
```

If user passed `columns=[]`, this would override that with `list(features.schema["properties"])`, which I don't think was the intent.

Reply:
This is tested, so it seems to work fine. Looking into why, this is because in such a case `list(features.schema["properties"])` will also return `[]`, because we opened the file using fiona with the relevant `include_fields` keyword ensuring fiona knows we are not reading any columns.